src/protocols/output_management.rs

```diff
@@ -0,0 +1,791 @@
+    }
+}
+
+impl<D> Dispatch<ZwlrOutputManagerV1, (), D> for OutputManagementManagerState
+where
+    D: GlobalDispatch<ZwlrOutputManagerV1, OutputManagementManagerGlobalData>,
+    D: Dispatch<ZwlrOutputManagerV1, ()>,
+    D: Dispatch<ZwlrOutputHeadV1, String>,
+    D: Dispatch<ZwlrOutputConfigurationV1, OutputConfigurationData>,
+    D: Dispatch<ZwlrOutputConfigurationHeadV1, OutputConfigurationHeadState>,
+    D: Dispatch<ZwlrOutputModeV1, ()>,
+    D: OutputManagementHandler,
+    D: 'static,
+{
+    fn request(
+        state: &mut D,
+        client: &Client,
+        _manager: &ZwlrOutputManagerV1,
+        request: zwlr_output_manager_v1::Request,
+        _data: &(),
+        _display: &DisplayHandle,
+        data_init: &mut DataInit<'_, D>,
+    ) {
+        match request {
+            zwlr_output_manager_v1::Request::CreateConfiguration { id, serial } => {
+                let new_out = state.output_management_state().current_out.clone();
+                let g_state = state.output_management_state();
+                let conf = data_init.init(
+                    id,
+                    Mutex::new(OutputConfigurationDataInner { serial, new_out }),
+                );
+                if let Some(client_data) = g_state.clients.get_mut(&client.id()) {
+                    if serial != g_state.serial {
+                        conf.cancelled();
+                    }
+                    client_data.confs.insert(conf);
+                } else {
+                    error!("CreateConfiguration: missing client data");
+                }
+            }
+            zwlr_output_manager_v1::Request::Stop => {
+                let clients = &mut state.output_management_state().clients;
+                clients.get(&client.id()).map(|c| c.manager.finished());
+                clients.remove(&client.id());
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl<D> Dispatch<ZwlrOutputConfigurationV1, OutputConfigurationData, D>
+    for OutputManagementManagerState
+where
+    D: GlobalDispatch<ZwlrOutputManagerV1, OutputManagementManagerGlobalData>,
+    D: Dispatch<ZwlrOutputManagerV1, ()>,
+    D: Dispatch<ZwlrOutputHeadV1, String>,
+    D: Dispatch<ZwlrOutputConfigurationV1, OutputConfigurationData>,
+    D: Dispatch<ZwlrOutputConfigurationHeadV1, OutputConfigurationHeadState>,
+    D: Dispatch<ZwlrOutputModeV1, ()>,
+    D: OutputManagementHandler,
+    D: 'static,
+{
+    fn request(
+        state: &mut D,
+        client: &Client,
+        conf: &ZwlrOutputConfigurationV1,
+        request: zwlr_output_configuration_v1::Request,
+        conf_data: &OutputConfigurationData,
+        _display: &DisplayHandle,
+        data_init: &mut DataInit<'_, D>,
+    ) {
+        let g_state = state.output_management_state();
+        let outdated = conf_data.lock().unwrap().serial != g_state.serial;
+        match request {
+            zwlr_output_configuration_v1::Request::EnableHead { id, head } => {
```

Comment:
`It is a protocol error to send two enable_head or disable_head requests with the same head.` should I check it ?

Reply:
Yeah

Reply:
Because `It is a protocol error to omit a head in a configuration.` The initial configuration is just a copy of the old one, we some head enabled. 
So i guess i guess with the actual implementation, it will succeed and the client will be able to control the same output from 2 different objects.

Reply:
Keeping the heads enable allows to keep the `LogicalOutput` since there us no default logic for this (https://github.com/YaLTeR/niri/pull/413#discussion_r1658994165) so it's not really possible to now if and head as already been enable. Or i can put a boolean.